File: examples/HelloWorldJava/src/java/com/example/helloworld/MainActivity.java
package com.example.helloworld;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends Activity {

    public native String getMessage();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView text = findViewById(R.id.text);

        text.setText("Hello World!");
    }
}

File: examples/Native_Activity_Basic_Kotlin/src/kotlin/com/example/nativec/kt/MainActivity.kt
package com.example.nativec.kt

import android.app.Activity
import android.os.Bundle
import android.widget.TextView

class MainActivity : Activity() {

    companion object {
        init {
            System.loadLibrary("native")
        }
    }

    external fun getMessage(): String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        val text: TextView = findViewById(R.id.text)

        text.text = getMessage()
    }
}